packages/cli/mock/src/runMockServer.ts

```diff
@@ -37,33 +36,95 @@ export class MockServer {
         this.app.use(express.json({ limit: "50mb", strict: false }));
         this.port = port;
 
+        // Map of endpoint path to map of HTTP method to endpoints that fall into both
+        const endpointGroups: Map<string, Map<string, HttpEndpoint[]>> = new Map();
+
         for (const service of Object.values(ir.services)) {
             for (const endpoint of service.endpoints) {
                 const endpointPath = getFullPathForEndpoint(endpoint);
                 context.logger.debug(`Registering ${endpoint.method} ${endpointPath} ...`);
-                switch (endpoint.method) {
+
+                const existingGroup = endpointGroups.get(endpointPath) ?? new Map();
+                const existingEndpoints = existingGroup.get(endpoint.method) ?? [];
+
+                endpointGroups.set(endpointPath, existingGroup.set(endpoint.method, [...existingEndpoints, endpoint]));
+            }
+        }
+
+        const listGroups = Array.from(endpointGroups);
+        const sortedEndpoints = listGroups.sort(this.sortExpressPaths);
```

Comment:
nit: It might be worth making a private function devoted to creating the `sortedEndpoints` variable (i.e. the `for` loop above).